sharding-core/sharding-core-preprocessor/src/test/java/org/apache/shardingsphere/core/optimize/segment/insert/InsertValueContextTest.java

```diff
@@ -0,0 +1,155 @@
+
+        Method calculateParametersCountMethod = InsertValueContext.class.getDeclaredMethod("calculateParametersCount", Collection.class);
+        calculateParametersCountMethod.setAccessible(true);
+        int calculateParametersCountResult = (int) calculateParametersCountMethod.invoke(insertValueContext, new Object[] {assignments});
+        assertThat(insertValueContext.getParametersCount(), is(calculateParametersCountResult));
+
+        Method getValueExpressionsMethod = InsertValueContext.class.getDeclaredMethod("getValueExpressions", Collection.class);
+        getValueExpressionsMethod.setAccessible(true);
+        List<ExpressionSegment> getValueExpressionsResult = (List<ExpressionSegment>) getValueExpressionsMethod.invoke(insertValueContext, new Object[] {assignments});
+        assertThat(insertValueContext.getValueExpressions(), is(getValueExpressionsResult));
+
+        Method getParametersMethod = InsertValueContext.class.getDeclaredMethod("getParameters", new Class[]{List.class, int.class});
+        getParametersMethod.setAccessible(true);
+        List<Object> getParametersResult = (List<Object>) getParametersMethod.invoke(insertValueContext, new Object[] {parameters, parametersOffset});
+        assertThat(insertValueContext.getParameters(), is(getParametersResult));
+    }
+
+    @Test
+    public void assertGetValueWhenParameterMarker() {
+        Collection assignments = makeParameterMarkerExpresstionSegment();
+        String parameterValue = "test";
+        List parameters = Lists.newArrayList(parameterValue);
+        int parametersOffset = 0;
+
+        InsertValueContext insertValueContext = new InsertValueContext(assignments,parameters, parametersOffset);
+        Object result = insertValueContext.getValue(0);
+
+        assertThat((String)result, is(parameterValue));
+    }
+
+    private Collection<ParameterMarkerExpressionSegment> makeParameterMarkerExpresstionSegment() {
+        ParameterMarkerExpressionSegment parameterMarkerExpressionSegment = new ParameterMarkerExpressionSegment(0, 10, 5);
+        return Lists.newArrayList(parameterMarkerExpressionSegment);
+    }
+
+    @Test
+    public void assertGetValueWhenLiteralExpressionSegment() {
+        Object literalObject = new Object();
+        Collection assignments = makeLiteralExpressionSegment(literalObject);
+        List parameters = Lists.newArrayList();
+        InsertValueContext insertValueContext = new InsertValueContext(assignments,parameters, 0);
+        Object result = insertValueContext.getValue(0);
+
+        assertThat(result, is(literalObject));
+    }
+
+    private Collection<LiteralExpressionSegment> makeLiteralExpressionSegment(Object literalObject) {
+        LiteralExpressionSegment parameterMarkerExpressionSegment = new LiteralExpressionSegment(0, 10, literalObject);
+        return Lists.newArrayList(parameterMarkerExpressionSegment);
```

Comment:
If only single unit, use Collections.singletonList instead of Lists.newArrayList